### PR TITLE
Getting Node info for external API's by Http Request

### PR DESCRIPTION
**Pull request by agoz17**
I developed the http request structure to get node info as JSON. After executing runserver command http get can be tested via  this command : 

''http http://127.0.0.1:8000/nodes/rate_book/' .
(You should have rate_book intent in Rate node to get right node in JSON form)

http request can be made for a specific node will be on this url:  http://127.0.0.1:8000/nodes/(intent_of_node)/

We need convention while naming intents of the nodes. If we are going to match nodes from their intents for the http requests . We need to use alphanumeric convention to match the regex i have written. We can change this structure, but i suggest that it should be alphanumeric since it is the easiest way. If there is objection about it we can discuss it here since this code will be dependent to this  decision.

All nodes can be reached as JSON from this url also:  http://127.0.0.1:8000/nodes

I put this function while working on the code but if we decide that it is unnecessary, we can remove it.

By the way I forgot to add to how to install http cli app. For Linux and Mac 'pip install httpie' command would be sufficient to make http requests from your terminal with http command.

### CODE REVIEW

File: bookbotproject/conversationtree/views.py [hunk 1]
```diff
@@ -2,5 +2,55 @@
...
+from rest_framework.renderers import JSONRenderer
+from rest_framework.parsers import JSONParser
+from conversationtree.models import Node
+from serializers import NodeSerializer
 
 # Create your views here.
+@csrf_exempt
+def node_list(request):
+    """
+    List all code snippets, or create a new snippet.
```
Comment (theTytoAlba) [inline bookbotproject/conversationtree/views.py]: Please fix the comment:
"List all nodes or create a new node"

File: bookbotproject/conversationtree/views.py [hunk 2]
```diff
@@ -2,5 +2,55 @@
...
+        if serializer.is_valid():
+            serializer.save()
+            return JsonResponse(serializer.data, status=201)
+        return JsonResponse(serializer.errors, status=400)
+
+
+@csrf_exempt
+def nodes_detail(request, pk):
+    """
+    Retrieve, update or delete a code snippet.
```
Comment (theTytoAlba) [inline bookbotproject/conversationtree/views.py]: Please fix comment:
"Retrieve, update or delete a node"

**Review by theTytoAlba — COMMENTED** [review]
Hi! Thank you for your contribution. I think this is going to be useful when we are getting nodes while following the conversation tree.
Only change we need for this branch is to update the comments.

As a next step, after we merge this, can you please create a function to get the children of the given node? Currently we have a function nodes_detail and we can get the node with that function. We should have another function get_node_children which takes in the same parameters and returns the children nodes. You can check out this function here, mptt has this functionality by default:
https://django-mptt.github.io/django-mptt/models.html#get-children

Thank you!

**Comment by theTytoAlba** [comment]
Amazing job, thank you for the quick response!

**Comment by agoz17** [comment]
I have updated the things you've noticed. Thanks for reaction Irmak. @theTytoAlba And I'm going to create the issue for the function you mentioned and assign it to myself.

**Pull request merged**